pydm/widgets/line_edit.py

```diff
@@ -270,6 +271,19 @@ def set_display(self):
 
         self.setText(self._display)
 
+    def focusInEvent(self, event: QFocusEvent) -> None:
+        """
+        Checks to see if the line edit has actually connected before assigning active window or tab focus to it. PyQt
+        will automatically give tab focus to the first tab-enabled widget it can on display load. But for this widget,
+        this behavior can lead to a race condition where if the widget is given focus before the PV connects, then
+        the widget never loads the initial text from the PV.
+        """
+        if not self._connected and (event.reason() == Qt.ActiveWindowFocusReason or event.reason() == Qt.TabFocusReason):
```

Comment:
I thought this looked super reasonable, so I tested it out. I had a lot of trouble reproducing this again, but after enough tries the network has slowed down enough (?) and I saw the issue.

Then, I swapped in this patch as a test, but I still have the same issue.
I think the problem is that `self._connected` here actually defaults to `True`, so when we hit this during the first initial setup we skip past this `clearFocus` call entirely. `self._connected` only goes `False` once we've timed out (I think?) but if we're right on the edge where we are in the process of connecting but haven't loaded in a value yet then it remains `True`.

https://github.com/slaclab/pydm/blob/b16e89ebea218604437bb767c3d77f379958854b/pydm/widgets/base.py#LL613C31-L613C31

Maybe instead of relying on init-time connected behavior, we could instead check if `self._display` has ever been set to a non-`None` value? There are probably other good solutions too.

Reply:
Shoot, thanks for testing and looking into it, I will do the check with `self._display` instead.